core/store/src/trie/update.rs

```diff
@@ -20,6 +20,10 @@ pub struct TrieUpdate {
     prospective: BTreeMap<Vec<u8>, Option<Vec<u8>>>,
 }
 
+/// For each prefix, the value is a <key, value> map that records the changes in the
```

Comment:
@mikhailOK  Why are we using BTreeMap in TrieUpdate in the first place? It seems like having an ordering is unnecessary, and HashMap is O(1) instead of O(log n).